From 1ffe04d0f2ef3c7737f1f9667cfcdd6c4239bbbe Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Wed, 17 Apr 2019 23:58:55 +0200
Subject: [PATCH] FindPCAP.cmake: fix static build

When statically built, pcap can depends on other libraries such as
-lnl-3. Add a call to pkg-config to find them and don't overwrite
PCAP_LIBRARIES with PCAP_LIBRARY (use APPEND instead)

Fixes:
 - http://autobuild.buildroot.org/results/277a24ad543a8f54cc8afde4f359f5d77b42eb7f

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: not sent yet (waiting feedback on first patch)]
---
 cmake/modules/FindPCAP.cmake | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/cmake/modules/FindPCAP.cmake b/cmake/modules/FindPCAP.cmake
index dfd80415f6..826ccfbdcc 100644
--- a/cmake/modules/FindPCAP.cmake
+++ b/cmake/modules/FindPCAP.cmake
@@ -9,6 +9,11 @@
 include( FindWSWinLibs )
 FindWSWinLibs( "WpdPack" "PCAP_HINTS" )
 
+if (NOT WIN32)
+  find_package(PkgConfig)
+  pkg_search_module( PCAP libpcap )
+endif()
+
 # The 64-bit wpcap.lib is under /x64
 set ( _PLATFORM_SUBDIR "" )
 if( WIN32 AND "${WIRESHARK_TARGET_PLATFORM}" STREQUAL "win64" )
@@ -37,7 +42,7 @@ find_package_handle_standard_args( PCAP DEFAULT_MSG PCAP_LIBRARY PCAP_INCLUDE_DI
 
 if( PCAP_FOUND )
   set( PCAP_INCLUDE_DIRS ${PCAP_INCLUDE_DIR} )
-  set( PCAP_LIBRARIES ${PCAP_LIBRARY} )
+  list( APPEND PCAP_LIBRARIES ${PCAP_LIBRARY} )
 else()
   set( PCAP_INCLUDE_DIRS )
   set( PCAP_LIBRARIES )
-- 
2.20.1

